--- geonode/base/models.py      2016-04-30 19:40:39.812000000 -0400
+++ geonode/base/models_patched.py      2016-04-30 19:41:02.902000000 -0400
@@ -521,6 +521,8 @@
             legends_link = self.link_set.get(name='Legend')
         except Link.DoesNotExist:
             return None
+        except Link.MultipleObjectsReturned, e:
+            return None
         else:
             return legends_link
